src/openvic-simulation/politics/NationalFocus.cpp

```diff
@@ -0,0 +1,107 @@
+
+NationalFocus::NationalFocus(
+	std::string_view new_identifier,
+	std::string_view new_icon,
+	NationalFocusGroup const* new_group,
+	ModifierValue&& new_modifiers,
+	pop_promotion_map_t&& new_encouraged_promotion,
+	party_loyalty_map_t&& new_encouraged_loyalty,
+	production_map_t&& new_encouraged_production
+) : HasIdentifier { new_identifier },
+	icon { new_icon },
+	group { new_group },
+	modifiers { new_modifiers },
+	encouraged_promotion { new_encouraged_promotion },
+	encouraged_loyalty { new_encouraged_loyalty },
+	encouraged_production { new_encouraged_production } {}
+
+NationalFocusManager::NationalFocusManager() : national_focus_groups { "national focus groups" }, national_foci { "national foci" } {}
+
+inline bool NationalFocusManager::add_national_focus_group(std::string_view identifier) {
+	return national_focus_groups.add_item({ identifier });
+}
+
+inline bool NationalFocusManager::add_national_focus(
+	std::string_view identifier,
+	std::string_view icon,
+	NationalFocusGroup const* group,
+	ModifierValue&& modifiers,
+	NationalFocus::pop_promotion_map_t&& encouraged_promotion,
+	NationalFocus::party_loyalty_map_t&& encouraged_loyalty,
+	NationalFocus::production_map_t&& encouraged_production
+) {
+	return national_foci.add_item({ identifier, icon, group, std::move(modifiers), std::move(encouraged_promotion), std::move(encouraged_loyalty), std::move(encouraged_production) });
+}
+
+bool NationalFocusManager::load_national_foci_file(PopManager const& pop_manager, IdeologyManager const& ideology_manager, GoodManager const& good_manager, ModifierManager const& modifier_manager, ast::NodeCPtr root) {
+	bool ret = expect_dictionary([this](std::string_view identifier, ast::NodeCPtr node) -> bool {
+		return add_national_focus_group(identifier);
+	})(root);
+	lock_national_focus_groups();
+
+	ret &= expect_dictionary([this, &pop_manager, &ideology_manager, &good_manager, &modifier_manager](std::string_view group_identifier, ast::NodeCPtr node) -> bool {
+		NationalFocusGroup const* current_group;
+		bool ret = expect_national_focus_group_str(assign_variable_callback_pointer(current_group))(group_identifier);
+		
+		ret &= expect_dictionary([this, &current_group, &pop_manager, &ideology_manager, &good_manager, &modifier_manager](std::string_view identifier, ast::NodeCPtr node) -> bool {
+			std::string_view icon;
+			ModifierValue modifiers;
+			NationalFocus::pop_promotion_map_t promotions;
+			NationalFocus::party_loyalty_map_t loyalties;
+			NationalFocus::production_map_t production;
+
+			Ideology const* last_specified_ideology = nullptr; // weird, I know
+
+			bool ret = modifier_manager.expect_modifier_value_and_keys_and_default(
+				move_variable_callback(modifiers),
+				[&promotions, &pop_manager, &production, &good_manager, &modifiers, &modifier_manager](std::string_view key, ast::NodeCPtr value) -> bool {
+					if (pop_manager.has_pop_type_identifier(key)) {
```

Comment:
Under the hood this searches for the pop type pointer and checks it's not null, if you get the pointer and do the check yourself you can avoid an unnecessary second search. (Same thing for the goods check beneath this)